test/model.querying.test.js

```diff
@@ -2399,6 +2399,19 @@ describe('model: querying:', function() {
       });
     });
 
+    it('removes the __v property if versionKey: false is set (gh-8934)', function(done) {
+      const title = 'Wooooot ' + random();
+
+      const post = new BlogPostB();
+      post.set('title', title);
+      post.save(function(err) {
+        BlogPostB.find({ title: title }).lean({ versionKey: false }).exec(function(err, docs) {
```

Comment:
This can be simplified by using async/await, and `findOne` instead of `find`
```js
const title = 'Wooooot ' + random();
const post = await BlogPostB.create({ title });
const foundPost= await BlogPostB.find({ title }).lean({  versionKey: false });
assert.strictEqual(foundPost.__v, undefined);
```